privmx-endpoint/src/main/cpp/modules/InboxApi.cpp

```diff
@@ -663,58 +663,116 @@ Java_com_simplito_java_privmx_1endpoint_modules_inbox_InboxApi_closeFile(
+                }
+
+                auto subscription_ids_c = getInboxApi(ctx, thiz)->subscribeFor(
+                        subscription_queries_c);
+
+                jobject javaArrayList = env->NewObject(arrayListCls, initMID);
+                for (auto &id_str : subscription_ids_c) {
+                    jstring java_id_str = ctx->NewStringUTF(id_str.c_str());
+                    env->CallBooleanMethod(javaArrayList, addToListMID, java_id_str);
+                    env->DeleteLocalRef(java_id_str);
+                }
+                env->DeleteLocalRef(arrayListCls);
+
+                return javaArrayList;
+            }
+    );
+
+    if (ctx->ExceptionCheck()) {
+        return nullptr;
+    }
+    return result;
 }
 
 extern "C"
 JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_inbox_InboxApi_subscribeForEntryEvents(
+Java_com_simplito_java_privmx_1endpoint_modules_inbox_InboxApi_unsubscribeFrom(
         JNIEnv *env,
         jobject thiz,
-        jstring inbox_id
+        jobject subscription_ids
 ) {
     JniContextUtils ctx(env);
-    if (ctx.nullCheck(inbox_id, "Inbox ID")) {
+    if (ctx.nullCheck(subscription_ids, "Subscription IDs for InboxApi")) {
         return;
     }
-    ctx.callVoidEndpointApi([&ctx, &thiz, &inbox_id]() {
-        getInboxApi(ctx, thiz)->subscribeForEntryEvents(
-                ctx.jString2string(inbox_id)
-        );
+
+    ctx.callVoidEndpointApi([&ctx, &env, &thiz, &subscription_ids]() {
+        auto subscription_ids_arr = ctx.jObject2jArray(subscription_ids);
+        auto subscription_ids_c = std::vector<std::string>();
+
+        int length = ctx->GetArrayLength(subscription_ids_arr);
+        for (int i = 0; i < length; i++) {
+            jobject arrayElement = ctx->GetObjectArrayElement(subscription_ids_arr, i);
+            subscription_ids_c.push_back(ctx.jString2string((jstring) arrayElement));
+        }
+
+        getInboxApi(ctx, thiz)->unsubscribeFrom(subscription_ids_c);
     });
 }
 
+
 extern "C"
-JNIEXPORT void JNICALL
-Java_com_simplito_java_privmx_1endpoint_modules_inbox_InboxApi_unsubscribeFromEntryEvents(
+JNIEXPORT jstring JNICALL
+Java_com_simplito_java_privmx_1endpoint_modules_inbox_InboxApi_buildSubscriptionQuery(
         JNIEnv *env,
         jobject thiz,
-        jstring inbox_id
+        jlong eventType,
+        jlong selectorType,
+        jstring selectorId
 ) {
     JniContextUtils ctx(env);
-    ctx.callVoidEndpointApi([&ctx, &thiz, &inbox_id]() {
-        getInboxApi(ctx, thiz)->unsubscribeFromEntryEvents(
-                ctx.jString2string(inbox_id)
-        );
-    });
+    if (ctx.nullCheck(selectorId, "SelectorID")) {
+        return nullptr;
+    }
+
+    jstring result = nullptr;
+    ctx.callResultEndpointApi<jstring>(
+            &result,
+            [&ctx, &env, &thiz, &eventType, &selectorType, &selectorId]() {
+                auto c_eventType = parseInboxEventType(ctx, (long) eventType);
```

Comment:
you can easy cast long value to enum like this
```suggestion
                auto c_eventType = static_cast<inbox::EventType>(eventType);
```